### ISSUE TITLE
Built-in binaries are being detected as unlisted

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [webpro/knip#116]
Given this file: https://github.com/navin-moorthy/template-typescript-node-package/blob/8db8d75bf8288058a0a51c894235d0a2d282c497/script/setup.js

...with lines like this:

```js
const { stdout } = await $`npm whoami`.pipeStdout($`cat`);
```

```js
await $`gh auth status`;
```

Knip is reporting binaries as unlisted:

```plaintext
Unlisted dependencies (2)
bin:cat  script/setup.js
bin:gh   script/setup.js
```

This can be solved with `ignoreBinaries`:

```json
{
	"ignoreBinaries": ["cat", "gh"],
}
```

I can empathize with the `gh` binary being mentioned. But `cat` comes with Linux & Mac. Could we add some kind of option for including all OS-provided scripts? Maybe `"ignoreBinaries": ["<linux>"]` or something like that?

**Comment by webpro** [comment]
Yes, it's a bit tricky. Although `cat` is obviously a candidate to add, since that npm package is old. But name clashes between packages and built-in binaries is something to avoid. Honestly I expected more issues in this field, it's basically a mine field. That's why I'm a bit hesitant to add too much. Here's the current list (I'll definitely add `cat`):

https://github.com/webpro/knip/blob/2ff0799ea332e147bb9906fa64d8028824a6f71a/src/constants.ts#L16-L36

It's an interesting idea, but having something like `<linux>` might be unclear for people and/or CI environments with different OSs. It might give the impression that it would behave different or feel off on Windows.

**Comment by JoshuaKGoldberg** [comment]
Yeah the different OS feel is interesting. `cat` _is_ an unlisted binary on Windows...

**Comment by webpro** [comment]
Haha, it works the other way around too indeed. Perhaps best to have a single list for now and see how this evolves. All the input and ideas are very welcome and much appreciated.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.8.0. See [Release 2.8.0](https://github.com/webpro/knip/releases/tag/2.8.0) for release notes._
